tests/DataCache.test.ts

```diff
@@ -131,4 +131,88 @@ describe("DataCache", () => {
     const cache = new DataCache<DisposableString>(3);
     expect(() => cache.insert("1", new DisposableString("A"), 4)).toThrowError(ANY_ERROR);
   });
+
+  it("reserves keys and prevents them from being evicted", () => {
+    const cache = new DataCache<DisposableString>(3);
+    cache.insert("1", new DisposableString("A"));
+    cache.insert("2", new DisposableString("B"));
+    cache.insert("3", new DisposableString("C"));
+    cache.setReservedKeys(new Set(["1", "2"]));
+    expect(cache.size).toBe(3);
+
+    // 3 should be evicted because 1 and 2 are reserved.
+    cache.insert("4", new DisposableString("D"));
+    expect(cache.size).toBe(3);
```

Comment:
why is the cache size still 3 here? it seems like you added another item?

Reply:
oh! I get it I think, the size remains constant and because you added a new one it overwrote "3" in order to add 4:D

Reply:
I'll add a comment on this! Sorry for the puzzle 😅